internal/scorer/scorer.go

```diff
@@ -103,10 +83,19 @@ func NameFromFilepath(filepath string) string {
 	// Get the name of the file used, without the path
 	f := path.Base(filepath)
 	ext := path.Ext(f)
-	// Strip the extension and convert to lowercase
-	f = strings.ToLower(strings.TrimSuffix(f, ext))
-	// Change any non-alphanumeric character into an underscore
-	f = regexp.MustCompile("[^a-z0-9_]").ReplaceAllString(f, "_")
+
+	modified := func(r rune) rune {
+		// Change any non-alphanumeric character into an underscore
+		if !unicode.IsDigit(r) && !unicode.IsLetter(r) {
```

Comment:
This condition does not match `"[^a-z0-0_]"` as it will preserve unicode letters in the filename in the output.

For example a filename of "configuración_básica.yaml" previously would have resulted in `configuraci_n_b_sica_score`, now it produces `configuración_básica_score` (arguably better).

The milestone 1 doc doesn't specify the encoding of field names, but does permit Unicode strings.

Given that this changes the behaviour it should be covered by a change in tests too.

Reply:
Which tests do you mean? I don't see any tests covering `NameFromFilepath()`, and I can't see any tests on any place it is being used.

Reply:
Oh, I misread your comment. I will write tests for it!

Reply:
I have written the tests.